external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/EmptyKafkaTupleListener.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.storm.kafka.spout;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.storm.spout.SpoutOutputCollector;
+import org.apache.storm.task.TopologyContext;
+
+public class EmptyKafkaTupleListener implements KafkaTupleListener {
+
+    @Override
+    public void open(Map<String, Object> conf, TopologyContext context, SpoutOutputCollector collector) { }
+
+    @Override
+    public void onEmit(List<Object> tuple, KafkaSpoutMessageId msgId) { }
+
+    @Override
+    public void onAck(KafkaSpoutMessageId msgId) { }
+
+    @Override
+    public void onPartitionsReassigned(KafkaSpoutMessageId msgId) { }
```

Comment:
I think this should take the list of assigned partitions as a parameter, and be called in the onPartitionsAssigned of the rebalance listener instead. That would allow implementations to delete any saved message ids that are for revoked partitions easily.